### ISSUE TITLE
java.lang.IndexOutOfBoundsException: Range [45, 45 + -15) out of bounds for length 30

### ISSUE THREAD

**Issue by ElAndree** [Minestom/Minestom#609]
Hey there,

sometimes my players are getting randomly disconnected with the following exception:
Is this a minestom bug or do you have a clue why this happens?

Thanks!
Kind regards,
Andre

```
java.lang.IndexOutOfBoundsException: Range [45, 45 + -15) out of bounds for length 30
    at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
    at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckFromIndexSize(Preconditions.java:82)
    at java.base/jdk.internal.util.Preconditions.checkFromIndexSize(Preconditions.java:361)
    at java.base/java.util.Objects.checkFromIndexSize(Objects.java:411)
    at java.base/java.nio.DirectByteBuffer.slice(DirectByteBuffer.java:255)
    at java.base/java.nio.DirectByteBuffer.slice(DirectByteBuffer.java:40)
    at net.minestom.server.network.player.PlayerSocketConnection.writeBufferSync(PlayerSocketConnection.java:388)
    at net.minestom.server.network.player.PlayerSocketConnection.lambda$write$2(PlayerSocketConnection.java:176)
    at org.jctools.queues.MessagePassingQueueUtil.drain(MessagePassingQueueUtil.java:53)
    at org.jctools.queues.MpUnboundedXaddArrayQueue.drain(MpUnboundedXaddArrayQueue.java:435)
    at org.jctools.queues.MpscUnboundedXaddArrayQueue.drain(MpscUnboundedXaddArrayQueue.java:28)
    at net.minestom.server.network.socket.Worker.run(Worker.java:50)
```

**Comment by TheMode** [comment]
At first glance seems to be on our side. But I will need a reproducer.

**Comment by emortaldev** [comment]
I can't reproduce, but I was just getting this issue
![image](https://user-images.githubusercontent.com/28760284/151408369-449163c6-58ac-4847-af1c-8b7f87e83356.png)

**Comment by ElAndree** [comment]
@emortaldev have you activated MojangAuth?

**Comment by emortaldev** [comment]
I have it activated, yes. Does turning it off fix it for you?

**Comment by TheMode** [comment]
Can you reproduce?

**Comment by emortaldev** [comment]
I haven't had the issue for a bit now which is a bit annoying but I'll update anyway

**Comment by iam4722202468** [comment]
I haven't had this issue recently either

**Comment by iam4722202468** [comment]
Update: I had this problem on f9080fa4f86b9ad196231e7ba631ebc293dba1b8

**Comment by TheMode** [comment]
Should be fixed by 2e02dfd4d8c9d865c9e0b1b5941060486f54607f
